ios/FluentUI/Badge Field/BadgeView.swift

```diff
@@ -187,11 +137,148 @@ open class BadgeView: UIView {
 
     @objc open var isSelected: Bool = false {
         didSet {
-            updateBackgroundColor()
-            updateLabelTextColor()
+            updateColors()
             updateAccessibility()
         }
     }
+    
+    private var _labelTextColor: UIColor?
+    open var labelTextColor: UIColor? {
+        get {
+            if let customLabelTextColor = _labelTextColor {
+                return customLabelTextColor
+            }
+            
+            switch style {
+                case .default:
+                    if let window = window {
+                        return Colors.primary(for: window)
+                    }
+                case .warning:
+                    return Colors.Badge.textWarning
+                case .error:
+                    return Colors.Badge.textError
+            }
+            return nil
+        }
+        set {
+            if labelTextColor != newValue {
+                _labelTextColor = newValue
+                updateColors()
+            }
+        }
+    }
+    
+    private var _selectedLabelTextColor: UIColor?
+    open var selectedLabelTextColor: UIColor {
+        get {
+            if let customSelectedLabelTextColor = _selectedLabelTextColor {
+                return customSelectedLabelTextColor
+            }
+            
+            switch style {
+                case .default:
+                    return Colors.Badge.textSelected
+                case .warning:
+                    return Colors.Badge.textWarningSelected
+                case .error:
+                    return Colors.Badge.textErrorSelected
+            }
+        }
+        set {
+            if selectedLabelTextColor != newValue {
+                _selectedLabelTextColor = newValue
+                updateColors()
+            }
+        }
+    }
+    
+    private var _disabledLabelTextColor: UIColor?
+    open var disabledLabelTextColor: UIColor? {
+        get {
+            if let customDisabledLabelTextColor = _disabledLabelTextColor {
+                return customDisabledLabelTextColor
+            }
+            return style == .default ? Colors.Badge.textDisabled : (isActive ? self.selectedLabelTextColor : self.labelTextColor)
```

Comment:
Why are we returning the selectedLabelTextColor or labelTextColor on the disabledLabelTextColor?

Reply:
Currently, there is only defined case for `disabledColor` when style is an `default`. 
If other styles like `warning` and `error` you fallback to their original preset colors. 

The relevant code before changes is [here](https://github.com/microsoft/fluentui-apple/blob/897aa9e980cb87a1fa46c28916e0720b085cb2cb/ios/FluentUI/Badge%20Field/BadgeView.swift#L119).

Reply:
return style == .default ? Colors.Badge.textDisabled : (**isSelected** ? self.selectedLabelTextColor : self.labelTextColor)